Watchman.DomainModel/Responses/ResponsesMessageService.cs

```diff
@@ -0,0 +1,185 @@
+                }
+
+                else
+                {
+                    _messagesService.SendEmbedMessage(title: null, description: null, messagesAboutDefaultResponses[i]);
+                }
+            }
+        }
+
+        private void PrintCustomResponses()
+        {
+            var messagesAboutCustomResponses = GetMessagesAboutCustomResponses();
+            for (int i = 0; i < messagesAboutCustomResponses.Count; ++i)
+            {
+                if (i == 0)
+                {
+                    _messagesService.SendEmbedMessage("Nadpisane responses:", description: null, messagesAboutCustomResponses[0]);
+                }
+
+                else
+                {
+                    _messagesService.SendEmbedMessage(title: null, description: null, messagesAboutCustomResponses[i]);
+                }
+            }
+
+            if (messagesAboutCustomResponses.Any() == false)
+            {
+                var message = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("-------------------------------", "```Obecnie jest brak nadpisanych responses!```")
+                };
+                _messagesService.SendEmbedMessage("Nadpisane responses:", description: null, message);
+            }
+        }
+
+        private void PrintAllResponses()
+        {
+            var messagesAboutAllResponses = GetMessagesAboutAllResponses();
+            for (int i = 0; i < messagesAboutAllResponses.Count; ++i)
+            {
+                if (i == 0)
+                {
+                    _messagesService.SendEmbedMessage("Wszystkie responses:", description: null, messagesAboutAllResponses[0]);
+                }
+
+                else
+                {
+                    _messagesService.SendEmbedMessage(title: null, description: null, messagesAboutAllResponses[i]);
+                }
+            }
+        }
+
+        private List<List<KeyValuePair<string, string>>> GetMessagesAboutDefaultResponses()
+        {
+            var defaultResponses = _responsesDatabase.GetResponsesFromBase()
+                                   .Where(x => x.IsDefault)
+                                   .ToList();
+            var messagesAboutDefaultResponses = new List<List<KeyValuePair<string, string>>>();
+
+            for (int i = 0; i < defaultResponses.Count; ++i)
+            {
+                if (i % MAX_NUMBER_OF_MESSAGE_FIELDS == 0)
+                {
+                    messagesAboutDefaultResponses.Add(new List<KeyValuePair<string, string>>(MAX_NUMBER_OF_MESSAGE_FIELDS));
+                }
+
+                var onEvent = defaultResponses[i].OnEvent;
+                var message = GetMessageWithoutFramesAndBold(defaultResponses[i].Message);
+                var oneField = new KeyValuePair<string, string>(onEvent, message);
+                messagesAboutDefaultResponses.Last().Add(oneField);
```

Comment:
Ja bym to zrobił tak:
typ zwracany metody: IEnumerable<List<KeyValuePair<string, string>>>
```var defaultResponses = new List<KeyValuePair<string, string>>();
for (int i = 0; i < defaultResponses.Count; i++)
{
    if (i % MAX_NUMBER_OF_MESSAGE_FIELDS == 0)
    {
        yield return defaultResponses;
        defaultReponses.Clear();
    }
    var onEvent = defaultResponses[i].OnEvent;
    var message = GetMessageWithoutFramesAndBold(defaultResponses[i].Message);
    var oneField = new KeyValuePair<string, string>(onEvent, message);
    defaultResponses.Add(oneField);
}

Reply:
1. Czemu preferujesz ten sposób? Jest wydajniejszy? Dla w sumie bez różnicy (nie licząc poniższych błędów)
2. Tutaj jest chyba błąd zamiast yield return defaultResponses raczej powinno być yield return messageAboutDefaultResponses, bo u mnie w kodzie defaultResponses to List<Response>, tak samo na końcu też powinno być messageAboutDefaultResponses oraz przed Clear. 
3. W zerowej iteracji jest wysyłana pusta lista, więc jeszcze trzeba drugi warunek dodać.

Reply:
Wiesz co, tak teraz myślę, że lepiej by było w ogóle logikę dzielenia na poszczególne wiadomości przenieść do nowego serwisu EmbedMessageSplittingService (możesz się wzorować na MessageSplittingService i MessagesService), bo co jeśli powiedzmy będziemy chcieli wysyłać listę roli jako embed message? Trzeba by kopiować wszystko na nowo
Drugi powód, to generowanie odpowiedzi nie powinno mieć nic wspólnego z dzieleniem wiadomości do wysłania